### PR TITLE
PX-622: new service to update Rockerbox client smart pixels

### PR DESCRIPTION
**Pull request by angelazhou-mntn**
[PX-662](https://mntn.atlassian.net/browse/PX-662)

**Description**
This is a new service which can update and delete Rockerbox client's smart pixels. 

Documentation: https://mntn.atlassian.net/wiki/spaces/ABT/pages/2214232065/smart-pixel-config-service

**Test Done**
- unit tests
- manual testings on the endpoints, functions, logging and etc.
- The service has been deployed to QA environment on 1/31. End-to-end testing was done by QA engineer. 

[PX-662]: https://mntn.atlassian.net/browse/PX-662?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/kotlin/com/steelhouse/smartpixelconfigservice/util/RbClientSpxUtil.kt
Comment (ahead123) [inline src/main/kotlin/com/steelhouse/smartpixelconfigservice/util/RbClientSpxUtil.kt:1]: @angelazhou-mntn - some of the rockerbox account id's will also contain underscores _ are we allowing for that as well here in the regex?

Comment (angelazhou-mntn) [reply src/main/kotlin/com/steelhouse/smartpixelconfigservice/util/RbClientSpxUtil.kt:1]: Yes. Alphanumeric, underscore and hyphen characters are allowed.

Comment (ahead123) [reply src/main/kotlin/com/steelhouse/smartpixelconfigservice/util/RbClientSpxUtil.kt:1]: Okay great thanks!

**Comment by angelazhou-mntn** [comment]
The pull request has been reviewed by Dako and everything looks good.

The service has been deployed to production on 2/6/2024 and everything looks good.

**Pull request merged**